test/unchecked.jl

```diff
@@ -0,0 +1,155 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+# unchecked integer arithmetic
+
+import Base.Unchecked:
+    unchecked_abs, unchecked_neg, unchecked_add, unchecked_sub,
+    unchecked_mul, unchecked_div, unchecked_rem, unchecked_fld,
+    unchecked_mod, unchecked_cld
+
+# Test return types
+for T in (Int8,Int16,Int32,Int64,Int128, UInt8,UInt16,UInt32,UInt64,UInt128)
+    z, o = T(0), T(1)
+    @test typeof(unchecked_neg(z)) === T
```

Comment:
Fair enough. (Though it doesn't look like `===` is commonly used for types in the existing tests currently.)